### PR TITLE
Box debug

### PR DESCRIPTION
**Pull request by TomasVotruba**
This fails with unclear error:

```bash
                                                                                                                                              
  The command "'../../vendor/bin/box' 'compile' '--config' '../../build/box.json'" failed.                                                    
                                                                                                                                              
  Exit Code: 102(Unknown error)                                                                                                               
                                                                                                                                              
  Working directory: /var/www/prefixer/tmp/build                                                                                              
                                                                                                                                              
  Output:                                                                                                                                     
  ================                                                                                                                            
                                                                                                                                              
      ____                                                                                                                                    
     / __ )____  _  __                                                                                                                        
    / __  / __ \| |/_/                                                                                                                        
   / /_/ / /_/ />  <                                                                                                                          
  /_____/\____/_/|_|                                                                                                                          
                                                                                                                                              
                                                                                                                                              
  Box version 3.4.0@4eeb285                                                                                                                   
                                                                                                                                              
                                                                                                                                              
   // Loading the configuration file "../../build/box.json".                                                                                  
                                                                                                                                              
   [ERROR] The configuration file is invalid.                                                                                                 
                                                                                                                                              
                                                                                                                                              
                                                                                                                                              
  Error Output:                                                                                                                               
  ================                                                                                                                            
                                                                                                                                              
  In Assertion.php line 2550:                                                                                                                 
                                                                                                                                              
    File "/var/www/prefixer/build/index.php" was expected to exist.                                                                           
```

### CODE REVIEW

File: src/Command/CompileCommand.php [hunk 1]
```diff
@@ -127,8 +123,10 @@ private function prepareSource(InputInterface $input): void
 
     private function buildPrefixedPhar(): void
     {
+        $this->processRunner->run(['cp', 'box.json', $this->buildDirectory . '/box.json']);
+
         $this->symfonyStyle->note('Building prefixed rector.phar');
-        $boxCommand = ['vendor/bin/box', 'compile', '--config', 'build/box.json'];
+        $boxCommand = ['vendor/bin/box', 'compile', '--config', 'box.json', '--working-dir', $this->buildDirectory];
```
Comment (theofidry) [inline src/Command/CompileCommand.php]: I don't think you should specify the box file here unless it's in a different directory than the working directory: Box will pick `box.json` or `box.json.dist` (in that order)

**Comment by theofidry** [comment]
>    File "/var/www/prefixer/build/index.php" was expected to exist.                                                                           

When the "autodiscovery" feature is enabled, i.e. Box tries to infer as much information available from the `composer.json` for the configuration, the first entry in the `bin` composer setting is being picked. Since you do not have one here, it fallsback on the traditional PHAR main script, `index.php`, resulting in a failure here since the main script in question cannot be found.

File: src/Command/CompileCommand.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Command/CompileCommand.php]: Could you simply send PR to this package or should I merge his to master first?

**Comment by TomasVotruba** [comment]
Just FYI, CI pass is false positive. The function autolaod is still the problem: https://travis-ci.com/rectorphp/compiler/builds/99439941#L791

image

**Pull request merged**